circleci/templates/base_config.yml

```diff
@@ -0,0 +1,142 @@
+            echo "export LINUX_MVN_RANDOM=-Djava.security.egd=file:/dev/./urandom" >> $BASH_ENV
+            echo "export COVERAGE_EXCLUSIONS=**/test/**/*,**/itests/**/*,**/*Test*,**/sdk/**/*,**/*.js,**/node_modules/**/*,**/jaxb/**/*,**/wsdl/**/*,**/nces/sws/**/*,**/*.adoc,**/*.txt,**/*.xml" >> $BASH_ENV
+            echo "export GITHUB_USERNAME=connexta" >> $BASH_ENV
+            echo "export GITHUB_REPONAME=ion-transformation" >> $BASH_ENV
+            echo "export GITHUB_KEY=ion-transformation-github-key" >> $BASH_ENV
+            source $BASH_ENV
+      - checkout
+      - run:
+          name: Building
+          command: |
+            mvn install -B -DskipTests $DISABLE_DOWNLOAD_PROGRESS_OPTS
+            mvn clean install -B -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/$CHANGE_TARGET $DISABLE_DOWNLOAD_PROGRESS_OPTS
+
+  test:
+    docker: *image
+    steps:
+      - run: *env
+      - checkout
+      - run:
+          name: Building
+          command: |
+            if [ $RELEASE = true ]
+            then
+              mvn -B -Dtag=$RELEASE_TAG -DreleaseVersion=$RELEASE_VERSION -DdevelopmentVersion=$NEXT_VERSION -Dgpg.secretKeyring=$ION_GPG_KEYRING -Dgpg.publicKeyring=$ION_GPG_KEYRING release:prepare
+              echo "export RELEASE_CONFIG = `git rev-parse HEAD`" >> $BASH_ENV
+            else
+              mvn clean install -B $DISABLE_DOWNLOAD_PROGRESS_OPTS
+            fi
+            mvn test
+      - run:
+          name: Save test results
+          command: |
+            mkdir -p ~/test-results/junit/
+            find . -type f -regex ".*/target/surefire-reports/.*xml" -exec cp {} ~/test-results/junit/ \;
+          when: always
+      - store_test_results:
+          path: ~/test-results
+      - store_artifacts:
+          path: ~/test-results/junit
+  owasp:
+    docker: *image
+    steps:
+      - run: *env
+      - checkout
+      - run:
+          name: Owasp Tests
+          command: |
+            if [ $CIRCLE_PR_NUMBER ]
+            then
+              mvn install -B -Powasp -DskipTests -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/$CHANGE_TARGET -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS
+            else
+              mvn install -B -Powasp -DskipTests -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS
+            fi
+            cp **/dependency-check-report.* /tmp/artifacts
+      - store_artifacts:
+          path: /tmp/artifacts
+  sonar:
+    docker: *image
+    steps:
+      - run: *env
+      - checkout
+      - run:
+          name: Sonar check
+          command: |
+            if [ $CIRCLE_BRANCH = master ]
+            then
+              if [ $RELEASE = true ]
+              then
+                git checkout $RELEASE_TAG
+              fi
+              mvn -q -B install sonar:sonar -Dsonar.host.url=https://sonarcloud.io -Dsonar.login=$SONAR_TOKEN  -Dsonar.organization=cx -Dsonar.projectKey=$GITHUB_REPONAME -Dsonar.exclusions=$COVERAGE_EXCLUSIONS $DISABLE_DOWNLOAD_PROGRESS_OPTS
```

Comment:
To get this data on GitHub you'll need to provide a few more flags:

`-Dsonar.pullrequest.key=${CIRCLE_PULL_REQUEST##*/} -Dsonar.pullrequest.branch=${CIRCLE_BRANCH} -Dsonar.pullrequest.base=master`

You can add this here, then test it in your straight-arrow repo by making a PR and running the workflow, and make sure it shows up on the bottom of the PR.